### PR TITLE
Make doc-build configurable

### PR DESCRIPTION
**Pull request by sgallagher**
Signed-off-by: Stephen Gallagher <sgallagh@redhat.com>

@KOLANICH

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -46,12 +46,17 @@ foreach cflag: test_cflags
...
 gobject = dependency('gobject-2.0')
 yaml = dependency('yaml-0.1')
-gtkdoc = dependency('gtk-doc')
+
+if with_docs
+    gtkdoc = dependency('gtk-doc')
+endif
 
 glib_prefix = dependency('glib-2.0').get_pkgconfig_variable('prefix')
 glib_docpath = join_paths(glib_prefix, 'share', 'gtk-doc', 'html')
```
Comment (mmathesius) [inline meson.build]: Should these lines and the tests for the presence of GLib and GObject documentation (a few lines below) also be included within an `if with_docs` block?

Comment (sgallagher) [reply meson.build]: Yes, they should. Fixing.

File: meson.build [hunk 2]
```diff
@@ -46,28 +46,34 @@ foreach cflag: test_cflags
   endif
 endforeach
 
+with_docs = get_option('with_docs')
```
Comment (mmathesius) [inline meson.build]: Totally optional: given the way things lay out now, it would make more sense to put this line right before the `if with_docs` block.

Comment (sgallagher) [reply meson.build]: Done

**Comment by mmathesius** [comment]
LTGM!

**Comment by KOLANICH** [comment]
Thank you!

**Pull request merged**